### DISCUSSION TITLE
SOAP Support?

### DISCUSSION THREAD

**Discussion by un1ck-andy** [usebruno/bruno#1378] [Q&A] [answered]
Hello,
Are there any plans to add support of SOAP requests?
Thanks!

**Comment by Estegar** [comment, marked as answer]
You actually can test SOAP request using Bruno :)

Set your request to post:
![image](https://github.com/usebruno/bruno/assets/110107206/ae9a52cc-9c89-4c60-94be-fd6bc267198e)

Then in body section:
![image](https://github.com/usebruno/bruno/assets/110107206/61f62a68-fb77-43ba-910f-7876982378d1)
Change to XML, paste your request xml in body, fill the values and you are good to go!
If you need xml -> json converter i could also provide something like that.

**Reply by un1ck-andy** [reply]
Wow, didn't know that! Thanks! 
P.S. I believe it would be beneficial to mention it somewhere! In fact we were pretty sure it is not possible and were looking for an additional tool just for SOAPs.

**Reply by Estegar** [reply]
Yeah, when i started with Burno, i had to dig a bit to find that info and setup.

**Reply by OpenOrangeData** [reply]
@helloanoop: +1 from me for mentioning prominently on your website that SOAP is possible, either with an official explanation or just a link here. (So far, SOAP isn't even mentioned in the documentation - although I would actually expect it even on the feature list on https://www.usebruno.com/pricing next to "REST") I was in the exact same position as @un1ck-andy. Some users might leave bruno for supposedly not supporting SOAP.

**Reply by chaithu8691** [reply]
@Estegar  .. Does bruno allow importing SOAP insomnia collection ? I tried to import the SOAP insomnia json file . The import was successful however I can't see the xml payload in the body ..

**Reply by OpenOrangeData** [reply]
@chaithu8691 - in case this could help: 
Additionally to the POST request with SOAP, in my case I had to add SoapAction=[link to URL]* (as key/value pair, two fields in the "header" tab of the request in Bruno) to the header, otherwise it didn't work.

*URL was more specific than the URL in the POST request, in my case it looked sth like this:
POST URL: example.com/xyz/Pull
URL for SoapAction: example.com/xyz/Pull/v1/pullDataXYZ

**Reply by andgeci** [reply]
is it possible to import wsdl with bruno?
